modin/test/test_internals.py

```diff
@@ -0,0 +1,27 @@
+import modin.pandas as pd
+
+
+def test_aligning_blocks():
+    # Test problem when modin frames have the same number of rows, but different
+    # blocks (partition.list_of_blocks). See #2322 for details
+    accm = pd.DataFrame(["-22\n"] * 162)
+    accm = accm.iloc[2:, :]
+    accm.reset_index(drop=True, inplace=True)
+    accm["T"] = pd.Series(["24.67\n"] * 145)
+
+    # see #2322 for details
+    repr(accm)
+
+
+def test_aligning_blocks_with_duplicated_index():
+    # Same problem as in `test_internal_error_2322` but with duplicated values in index.
+    data11 = [0, 1]
+    data12 = [2, 3]
+
+    data21 = [0]
+    data22 = [1, 2, 3]
+
+    df1 = pd.DataFrame(data11).append(pd.DataFrame(data12))
+    df2 = pd.DataFrame(data21).append(pd.DataFrame(data22))
+
+    repr(df1 - df2)
```

Comment:
Another observation:
```python
import modin.pandas as pd
import pandas

data11 = [0, 1]
data12 = [2, 3]

data21 = [0]
data22 = [1, 2, 3]

df1 = pd.DataFrame(data11).append(pd.DataFrame(data12))
df2 = pd.DataFrame(data21).append(pd.DataFrame(data22))
df1.columns = df2.columns = ['a']

pdf1 = pandas.DataFrame(data11).append(pandas.DataFrame(data12))
pdf2 = pandas.DataFrame(data21).append(pandas.DataFrame(data22))
pdf1.columns = pdf2.columns = ['a']

print(df1.a - df2) # give result
print(pdf1.a - pdf2) # failed with exception
```

Should we fail with exception in this case for aligning with pandas?

Reply:
Since this works: `print(pdf1 - pdf2)`, I would suggest that the inconsistency in the behavior of the pandas itself.

Can you create an appropriate issue in pandas to clarify the expected behavior?

Reply:
Yes, corresponded [issue](https://github.com/pandas-dev/pandas/issues/37663) in pandas was created.

Reply:
Thanks @prutskov .